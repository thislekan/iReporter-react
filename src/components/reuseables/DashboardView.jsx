/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/interactive-supports-focus */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import CreateReport from '../CreateReport.jsx';
import SingleReport from './SingleReport.jsx';
import ReportDetails from './ReportDetails.jsx';
import FilterComponent from '../search-and-filter/FilterComponent.jsx';
import SearchComponent from '../search-and-filter/SearchComponent.jsx';
import style from '../../styles/UserDashboard.css';
import promoImage from '../../media/promotion.svg';
import Loader from './Loader.jsx';
import EditReport from './EditReport.jsx';

const modalDefaultStyle = { content: { left: '20px', right: '20px', margin: '1rem 0' } };
const customizedStyle = {
  content: {
    left: '20px', right: '20px', padding: '10px', margin: '1rem 0',
  },
};

Modal.setAppElement('#app');

function DashboardView(props) {
  const {
    isLoading,
    closeCreateModal,
    handleChange,
    handleCreateIncident,
    disableFileInput,
    handleMedia,
    // videoBlob,
    imageBlobs,
    clearBlobs,
    openCreateModal,
    isDetailsModalOpen,
    closeDetailsModal,
    incidents,
    location,
    isCreateModalOpen,
    hideCreateButton,
    showCreateButton,
    fetchIncident,
    incident: details,
    mapUrl,
    isDeleteModalOpen,
    alterDeleteModal,
    deleteIncident,
    alterEditModal,
    isEditModalOpen,
    alterDisableComment,
    alterDisableLocation,
    disableComment,
    disableLocation,
    comment,
    incidentLocation,
  } = props;
  return (
    <div className={style.user__dashboard}>
      <div className={style['padded-body']}>
        {isLoading && <Loader isLoading={isLoading} />}
        <div>
          <Modal
            style={modalDefaultStyle}
            isOpen={isCreateModalOpen}
            onRequestClose={closeCreateModal}
            onAfterOpen={hideCreateButton}
            onAfterClose={hideCreateButton}
          >
            <div
              className={style.close__modal}
              onClick={closeCreateModal}
            >
              <FontAwesomeIcon
                icon='times-circle'
                style={{ height: '2rem', width: '2rem' }}
              />
            </div>
            <CreateReport
              handleCreateIncident={handleCreateIncident}
              handleChange={handleChange}
              handleMedia={handleMedia}
              disableFileInput={disableFileInput}
              // videoBlob={videoBlob}
              imageBlobs={imageBlobs}
              clearBlobs={clearBlobs}
            />
          </Modal>
          <Modal
            isOpen={isEditModalOpen}
            onRequestClose={alterEditModal}
            style={customizedStyle}
          >
            <div
              className={style.close__modal}
              onClick={alterEditModal}
            >
              <FontAwesomeIcon
                icon='times-circle'
                style={{ height: '2rem', width: '2rem' }}
              />
            </div>
            <EditReport
              incident={details}
              disableComment={disableComment}
              disableLocation={disableLocation}
              alterDisableComment={alterDisableComment}
              alterDisableLocation={alterDisableLocation}
              handleChange={handleChange}
              comment={comment}
              incidentLocation={incidentLocation}
            />
          </Modal>
        </div>
        <p id="welcome-text">Welcome. View your reports.</p>
        {
          (showCreateButton) ? null : <div
            className={style['create-modal']}
            id="create-modal"
            onClick={openCreateModal}
            role='button'
          >
            <img src={promoImage} alt="" />
          </div>}
        <Modal
          isOpen={isDeleteModalOpen}
          onRequestClose={alterDeleteModal}
          className={style['delete-modal-box']}
        >
          <div className={style['delete-prompt']}>
            <div className={style['prompt-body']}>
              <h3>Are you sure you want to delete this report?</h3>
              <div className={style['delete-btn-parent-div']}>
                <div>
                  <button
                    className={cx(style['alert-btns'], style['cancel-btn'])}
                    onClick={alterDeleteModal}
                  >
                    Cancel
                  </button>
                </div>
                <div>
                  <button
                    className={cx(style['delete-btn'], style['alert-btns'])}
                    onClick={deleteIncident}
                  >
                    Delete Report
                  </button>
                </div>
              </div>
            </div>
          </div>
        </Modal>
        <Modal
          style={customizedStyle}
          isOpen={isDetailsModalOpen}
          onRequestClose={closeDetailsModal}
        >
          <div
            className={style.close__modal}
            onClick={closeDetailsModal}
          >
            <FontAwesomeIcon
              icon='times-circle'
              style={{ height: '2rem', width: '2rem' }}
            />
          </div>
          <ReportDetails
            incident={details}
            mapUrl={mapUrl}
          />
        </Modal>
        <div className={style.body__content}>
          <div className={style.control}>
            <FilterComponent />
            <SearchComponent />
          </div>
          <div className="recently-created">
            {/* <h2 id="current-view-title"></h2> */}
            <div id="all-reports">
              <div
                className={style['list-of-reports']}
                id="incidents-view"
              >
                {
                  (!incidents.length) ? <p>
                    No incident reported yet
                      </p> : incidents.map(incident => <SingleReport
                      key={incident.id}
                      location={location}
                      id={incident.id}
                      createdOn={incident.createdOn}
                      status={incident.status}
                      type={incident.type}
                      incidentLocation={incident.location}
                      fetchIncident={fetchIncident}
                      alterDeleteModal={alterDeleteModal}
                      title={incident.title}
                      alterEditModal={alterEditModal}
                    />)
                }
              </div>
            </div>
          </div>
        </div>
      </div>
    </div >
  );
}

DashboardView.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  closeCreateModal: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleCreateIncident: PropTypes.func.isRequired,
  disableFileInput: PropTypes.bool.isRequired,
  handleMedia: PropTypes.func.isRequired,
  // videoBlob: PropTypes.array,
  imageBlobs: PropTypes.array,
  clearBlobs: PropTypes.func.isRequired,
  openCreateModal: PropTypes.func.isRequired,
  isDetailsModalOpen: PropTypes.bool.isRequired,
  closeDetailsModal: PropTypes.func.isRequired,
  incidents: PropTypes.array.isRequired,
  location: PropTypes.object.isRequired,
  isCreateModalOpen: PropTypes.bool.isRequired,
  hideCreateButton: PropTypes.func.isRequired,
  fetchIncident: PropTypes.func.isRequired,
  incident: PropTypes.object.isRequired,
  mapUrl: PropTypes.string,
  showCreateButton: PropTypes.bool.isRequired,
  isDeleteModalOpen: PropTypes.bool.isRequired,
  alterDeleteModal: PropTypes.func.isRequired,
  deleteIncident: PropTypes.func.isRequired,
  alterEditModal: PropTypes.func.isRequired,
  isEditModalOpen: PropTypes.bool.isRequired,
  alterDisableComment: PropTypes.func,
  alterDisableLocation: PropTypes.func,
  disableComment: PropTypes.bool,
  disableLocation: PropTypes.bool,
  comment: PropTypes.string,
  incidentLocation: PropTypes.string,
};

DashboardView.defaultProps = {
  mapUrl: '',
  alterDisableComment: f => f,
  alterDisableLocation: f => f,
  disableComment: true,
  disableLocation: true,
  incidentLocation: '',
  comment: '',
};

export default DashboardView;
