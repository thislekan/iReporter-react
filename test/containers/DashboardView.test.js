import '@babel/polyfill';
import React from 'react';
// import { shallow } from 'enzyme';
import setup from '../mock/setup';
import mockIncident from '../mock/incident';
// import ConnectedDashboard from '../../src/components/reuseables/DashboardView.jsx';
import DashboardView from '../../src/components/reuseables/DashboardView.jsx';

const mockFn = jest.fn();
const location = { pathname: '/user' };

describe('<UserForm />', () => {
  it('Should render all elements', () => {
    const wrapper = setup(
      <DashboardView
        location={location}
        incidents={[mockIncident]}
        isLoading={false}
        handleChange={mockFn}
        deleteIncident={mockFn}
        alterDeleteModal={mockFn}
        isDetailsModalOpen={false}
        closeDetailsModal={mockFn}
        fetchIncident={mockFn}
        alterEditModal={mockFn}
        incident={mockIncident}
        imageBlobs={[]}
        openModal={false}
        message=""
        resetState={mockFn}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    DashboardView.defaultProps.deleteIncident = mockFn;
    DashboardView.defaultProps.deleteIncident();
    const deleteBtn = wrapper.find('#delete_btn');
    deleteBtn.simulate('click', { deleteIncident: f => f });
    expect(mockFn).toHaveBeenCalled();
  });
});
