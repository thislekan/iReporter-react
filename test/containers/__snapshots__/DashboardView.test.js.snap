// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> Should render all elements 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <DashboardView
        alterDeleteModal={[MockFunction]}
        alterDisableComment={[Function]}
        alterDisableLocation={[Function]}
        alterDisableStatus={[Function]}
        alterEditModal={[MockFunction]}
        clearBlobs={[Function]}
        closeCreateModal={[Function]}
        closeDetailsModal={[MockFunction]}
        comment=""
        deleteIncident={[MockFunction]}
        disableComment={true}
        disableFileInput={false}
        disableLocation={true}
        disableStatus={true}
        fetchIncident={[MockFunction]}
        handleChange={[MockFunction]}
        handleCreateIncident={[Function]}
        handleMedia={[Function]}
        hideCreateButton={[Function]}
        imageBlobs={Array []}
        incident={
          Object {
            "comment": "jhdfhfdjk",
            "createdBy": "hjdjhdf",
            "createdOn": "48595940698459540",
            "creator": "jhdfkjdf",
            "id": "djfjfdkl-jeroier",
            "images": Array [
              "hi",
              "helo",
            ],
            "isLoading": false,
            "location": "sdjnkjs",
            "message": "kjfdj",
            "status": "draft",
            "title": "dhjfd",
            "type": "red-flag",
            "updatedOn": "19234895049",
            "videos": Array [
              "fdkfdn",
            ],
          }
        }
        incidentLocation=""
        incidentStatus=""
        incidentType=""
        incidents={
          Array [
            Object {
              "comment": "jhdfhfdjk",
              "createdBy": "hjdjhdf",
              "createdOn": "48595940698459540",
              "creator": "jhdfkjdf",
              "id": "djfjfdkl-jeroier",
              "images": Array [
                "hi",
                "helo",
              ],
              "isLoading": false,
              "location": "sdjnkjs",
              "message": "kjfdj",
              "status": "draft",
              "title": "dhjfd",
              "type": "red-flag",
              "updatedOn": "19234895049",
              "videos": Array [
                "fdkfdn",
              ],
            },
          ]
        }
        isCreateModalOpen={false}
        isDeleteModalOpen={false}
        isDetailsModalOpen={false}
        isEditModalOpen={false}
        isLoading={false}
        location={
          Object {
            "pathname": "/user",
          }
        }
        mapUrl=""
        message=""
        openCreateModal={[Function]}
        openModal={false}
        resetState={[MockFunction]}
        showCreateButton={false}
      >
        <div>
          <div>
            <div>
              <Modal
                ariaHideApp={true}
                bodyOpenClassName="ReactModal__Body--open"
                closeTimeoutMS={0}
                isOpen={false}
                onAfterClose={[Function]}
                onAfterOpen={[Function]}
                onRequestClose={[Function]}
                parentSelector={[Function]}
                portalClassName="ReactModalPortal"
                role="dialog"
                shouldCloseOnEsc={true}
                shouldCloseOnOverlayClick={true}
                shouldFocusAfterRender={true}
                shouldReturnFocusAfterClose={true}
              >
                <Portal
                  containerInfo={
                    <div
                      class="ReactModalPortal"
                    />
                  }
                >
                  <ModalPortal
                    ariaHideApp={true}
                    bodyOpenClassName="ReactModal__Body--open"
                    closeTimeoutMS={0}
                    defaultStyles={
                      Object {
                        "content": Object {
                          "WebkitOverflowScrolling": "touch",
                          "background": "#fff",
                          "border": "1px solid #ccc",
                          "borderRadius": "4px",
                          "bottom": "40px",
                          "left": "40px",
                          "outline": "none",
                          "overflow": "auto",
                          "padding": "20px",
                          "position": "absolute",
                          "right": "40px",
                          "top": "40px",
                        },
                        "overlay": Object {
                          "backgroundColor": "rgba(255, 255, 255, 0.75)",
                          "bottom": 0,
                          "left": 0,
                          "position": "fixed",
                          "right": 0,
                          "top": 0,
                        },
                      }
                    }
                    isOpen={false}
                    onAfterClose={[Function]}
                    onAfterOpen={[Function]}
                    onRequestClose={[Function]}
                    parentSelector={[Function]}
                    portalClassName="ReactModalPortal"
                    role="dialog"
                    shouldCloseOnEsc={true}
                    shouldCloseOnOverlayClick={true}
                    shouldFocusAfterRender={true}
                    shouldReturnFocusAfterClose={true}
                    style={
                      Object {
                        "content": Object {},
                        "overlay": Object {},
                      }
                    }
                  />
                </Portal>
              </Modal>
              <Modal
                ariaHideApp={true}
                bodyOpenClassName="ReactModal__Body--open"
                closeTimeoutMS={0}
                isOpen={false}
                onRequestClose={[MockFunction]}
                parentSelector={[Function]}
                portalClassName="ReactModalPortal"
                role="dialog"
                shouldCloseOnEsc={true}
                shouldCloseOnOverlayClick={true}
                shouldFocusAfterRender={true}
                shouldReturnFocusAfterClose={true}
              >
                <Portal
                  containerInfo={
                    <div
                      class="ReactModalPortal"
                    />
                  }
                >
                  <ModalPortal
                    ariaHideApp={true}
                    bodyOpenClassName="ReactModal__Body--open"
                    closeTimeoutMS={0}
                    defaultStyles={
                      Object {
                        "content": Object {
                          "WebkitOverflowScrolling": "touch",
                          "background": "#fff",
                          "border": "1px solid #ccc",
                          "borderRadius": "4px",
                          "bottom": "40px",
                          "left": "40px",
                          "outline": "none",
                          "overflow": "auto",
                          "padding": "20px",
                          "position": "absolute",
                          "right": "40px",
                          "top": "40px",
                        },
                        "overlay": Object {
                          "backgroundColor": "rgba(255, 255, 255, 0.75)",
                          "bottom": 0,
                          "left": 0,
                          "position": "fixed",
                          "right": 0,
                          "top": 0,
                        },
                      }
                    }
                    isOpen={false}
                    onRequestClose={[MockFunction]}
                    parentSelector={[Function]}
                    portalClassName="ReactModalPortal"
                    role="dialog"
                    shouldCloseOnEsc={true}
                    shouldCloseOnOverlayClick={true}
                    shouldFocusAfterRender={true}
                    shouldReturnFocusAfterClose={true}
                    style={
                      Object {
                        "content": Object {},
                        "overlay": Object {},
                      }
                    }
                  />
                </Portal>
              </Modal>
            </div>
            <p
              id="welcome-text"
            >
              Welcome. View your reports.
            </p>
            <div
              aria-hidden={true}
              id="create-modal"
              onClick={[Function]}
              role="button"
            >
              <img
                alt=""
                src={Object {}}
              />
            </div>
            <Modal
              ariaHideApp={true}
              bodyOpenClassName="ReactModal__Body--open"
              closeTimeoutMS={0}
              isOpen={false}
              onRequestClose={[MockFunction]}
              parentSelector={[Function]}
              portalClassName="ReactModalPortal"
              role="dialog"
              shouldCloseOnEsc={true}
              shouldCloseOnOverlayClick={true}
              shouldFocusAfterRender={true}
              shouldReturnFocusAfterClose={true}
            >
              <Portal
                containerInfo={
                  <div
                    class="ReactModalPortal"
                  />
                }
              >
                <ModalPortal
                  ariaHideApp={true}
                  bodyOpenClassName="ReactModal__Body--open"
                  closeTimeoutMS={0}
                  defaultStyles={
                    Object {
                      "content": Object {
                        "WebkitOverflowScrolling": "touch",
                        "background": "#fff",
                        "border": "1px solid #ccc",
                        "borderRadius": "4px",
                        "bottom": "40px",
                        "left": "40px",
                        "outline": "none",
                        "overflow": "auto",
                        "padding": "20px",
                        "position": "absolute",
                        "right": "40px",
                        "top": "40px",
                      },
                      "overlay": Object {
                        "backgroundColor": "rgba(255, 255, 255, 0.75)",
                        "bottom": 0,
                        "left": 0,
                        "position": "fixed",
                        "right": 0,
                        "top": 0,
                      },
                    }
                  }
                  isOpen={false}
                  onRequestClose={[MockFunction]}
                  parentSelector={[Function]}
                  portalClassName="ReactModalPortal"
                  role="dialog"
                  shouldCloseOnEsc={true}
                  shouldCloseOnOverlayClick={true}
                  shouldFocusAfterRender={true}
                  shouldReturnFocusAfterClose={true}
                  style={
                    Object {
                      "content": Object {},
                      "overlay": Object {},
                    }
                  }
                />
              </Portal>
            </Modal>
            <Modal
              ariaHideApp={true}
              bodyOpenClassName="ReactModal__Body--open"
              closeTimeoutMS={0}
              isOpen={false}
              onRequestClose={[MockFunction]}
              parentSelector={[Function]}
              portalClassName="ReactModalPortal"
              role="dialog"
              shouldCloseOnEsc={true}
              shouldCloseOnOverlayClick={true}
              shouldFocusAfterRender={true}
              shouldReturnFocusAfterClose={true}
            >
              <Portal
                containerInfo={
                  <div
                    class="ReactModalPortal"
                  />
                }
              >
                <ModalPortal
                  ariaHideApp={true}
                  bodyOpenClassName="ReactModal__Body--open"
                  closeTimeoutMS={0}
                  defaultStyles={
                    Object {
                      "content": Object {
                        "WebkitOverflowScrolling": "touch",
                        "background": "#fff",
                        "border": "1px solid #ccc",
                        "borderRadius": "4px",
                        "bottom": "40px",
                        "left": "40px",
                        "outline": "none",
                        "overflow": "auto",
                        "padding": "20px",
                        "position": "absolute",
                        "right": "40px",
                        "top": "40px",
                      },
                      "overlay": Object {
                        "backgroundColor": "rgba(255, 255, 255, 0.75)",
                        "bottom": 0,
                        "left": 0,
                        "position": "fixed",
                        "right": 0,
                        "top": 0,
                      },
                    }
                  }
                  isOpen={false}
                  onRequestClose={[MockFunction]}
                  parentSelector={[Function]}
                  portalClassName="ReactModalPortal"
                  role="dialog"
                  shouldCloseOnEsc={true}
                  shouldCloseOnOverlayClick={true}
                  shouldFocusAfterRender={true}
                  shouldReturnFocusAfterClose={true}
                  style={
                    Object {
                      "content": Object {},
                      "overlay": Object {},
                    }
                  }
                />
              </Portal>
            </Modal>
            <div>
              <div>
                <FilterFeature>
                  <div>
                    <label
                      htmlFor="select report"
                    >
                      Sort reported incidents by status
                    </label>
                    <select
                      id="sort-report"
                      name="select report"
                    >
                      <option
                        value="all"
                      >
                        All
                      </option>
                      <option
                        value="red-flag"
                      >
                        Red flags
                      </option>
                      <option
                        value="intervention"
                      >
                        Intervention
                      </option>
                      <option
                        value="draft"
                      >
                        Draft
                      </option>
                      <option
                        value="rejected"
                      >
                        Rejected
                      </option>
                      <option
                        value="resolved"
                      >
                        Resolved
                      </option>
                      <option
                        value="under-investigation"
                      >
                        Under Investigation
                      </option>
                    </select>
                  </div>
                </FilterFeature>
                <SearchFeature>
                  <div>
                    <label
                      htmlFor="filter list"
                    >
                      Search Incident
                    </label>
                    <select
                      id="filter-options"
                      name="filter list"
                    >
                      <option
                        value=""
                      >
                        Select option
                      </option>
                      <option
                        value="id"
                      >
                        Incident ID
                      </option>
                      <option
                        value="location"
                      >
                        Location
                      </option>
                    </select>
                    <input
                      id="filter-value"
                      name=""
                      placeholder="Please enter your search term here"
                      type="text"
                    />
                    <div>
                      <button>
                        Search
                      </button>
                      <button>
                        Clear Search
                      </button>
                    </div>
                  </div>
                </SearchFeature>
              </div>
              <div
                className="recently-created"
              >
                <div
                  id="all-reports"
                >
                  <div
                    id="incidents-view"
                  >
                    <SingleReport
                      alterDeleteModal={[MockFunction]}
                      alterEditModal={[MockFunction]}
                      createdOn="48595940698459540"
                      fetchIncident={[MockFunction]}
                      id="djfjfdkl-jeroier"
                      incidentLocation="sdjnkjs"
                      key="djfjfdkl-jeroier"
                      location={
                        Object {
                          "pathname": "/user",
                        }
                      }
                      status="draft"
                      title="dhjfd"
                      type="red-flag"
                    >
                      <div>
                        <div>
                          <div>
                            <div>
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="check-circle"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                              />
                              <h4>
                                Title: 
                              </h4>
                            </div>
                            <p>
                              dhjfd
                            </p>
                          </div>
                          <div>
                            <div>
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="check-circle"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                              />
                              <h4>
                                Type: 
                              </h4>
                            </div>
                            <p>
                              red-flag
                            </p>
                          </div>
                          <div>
                            <div>
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="check-circle"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                              />
                              <h4>
                                Date: 
                              </h4>
                            </div>
                            <p
                              id="incident-date"
                            >
                              Invalid date
                            </p>
                          </div>
                          <div>
                            <div>
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="check-circle"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                              />
                              <h4>
                                Status: 
                              </h4>
                            </div>
                            <p>
                              draft
                            </p>
                          </div>
                          <div>
                            <div>
                              <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="check-circle"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                              />
                              <h4>
                                Location: 
                              </h4>
                            </div>
                            <p>
                              sdjnkjs
                            </p>
                          </div>
                          <div>
                            <div>
                              <button
                                id="view_btn"
                                onClick={[Function]}
                              >
                                Details
                              </button>
                              <button
                                id="edit_btn"
                                onClick={[Function]}
                              >
                                Edit
                              </button>
                              <button
                                id="delete_btn"
                                onClick={[Function]}
                              >
                                Delete
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </SingleReport>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </DashboardView>
    </Router>
  </BrowserRouter>
</Provider>
`;
